common/src/cell.rs

```diff
@@ -219,8 +222,20 @@ where
 {
     /// Creates a new `SelfCell`.
     ///
-    /// The callback receives a pointer to the owned data. Note that a borrow to that data can only
-    /// safely be used to derive the object and **must not** leave the callback.
+    /// # Safety
```

Comment:
I wonder if you can just guarantee this via lifetimes somehow?

Reply:
Sadly, not. Any sort of guarantee would require self referential types.